File: frontend/movies-app/src/app/models/movie.ts
export interface Movie {
    pkMovies: number;
    name: string;
    gender: string;
    duration: string;
    fkDirector: number;
    directorName: string;
}

export interface SaveMovie {
    name: string;
    gender: string;
    duration: string;
    fkDirector: number;
}

File: backend/MoviesApi/Models/Movie.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MoviesApi.Models;

[Table("Movies")]
public class Movie
{
    [Key]
    [Column("PKMovies")]
    public int PKMovies { get; set; }

    [Required, StringLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required, StringLength(50)]
    public string Gender { get; set; } = string.Empty;

    [Required]
    public TimeSpan Duration { get; set; }

    public int FKDirector { get; set; }

    [ForeignKey(nameof(FKDirector))]
    public Director Director { get; set; } = null!;
}